### DISCUSSION TITLE
Is it possible to get information of the surrounding shell?

### DISCUSSION THREAD

**Discussion by nils-a** [spectreconsole/spectre.console#453] [Q&A] [answered]
I'd like to detect whether the program was called from `cmd`, `powershell` or something else. 

Is there something in spectre.console that can help with this?

**Comment by phil-scott-78** [comment, marked as answer]
Typically a shell is going to set the SHELL environmental variable. 

I gotta ask what your intentions are here. Anything besides just logging or displaying the value I'd recommend using the capabilities class which tries to abstract out what each console is capable of.

**Reply by nils-a** [reply]
I wanted to write some ready-to-copy code for the shell, something like: 

```
you need to call
set "FOO=bar"
```
on cmd. But on PowerShell it should rather be:

```
you need to call
$env:FOO="bar"
```

**Reply by phil-scott-78** [reply]
Ah, gotcha. I suspect checking the parent process name via `Process.GetCurrentProcess().Parent().ProcessName` might be the best way to go about it but you'd have to take account both powershell versions and have a good fallback for other shells or if it's being launched via another manner.

**Reply by nils-a** [reply]
Thanks. Will have a look at it. 👍
